pkg/controller/utils/utils.go

```diff
@@ -779,3 +783,54 @@ func GetDNSServiceName(provider operatorv1.Provider) types.NamespacedName {
 	}
 	return kubeDNSServiceName
 }
+
+// MonitorConfigMap starts a goroutine which exits if the given configmap's data is changed.
+func MonitorConfigMap(cs kubernetes.Interface, name string, data map[string]string) error {
+	informer := cache.NewSharedInformer(
+		cache.NewListWatchFromClient(
+			cs.CoreV1().RESTClient(),
+			"configmaps",
+			common.OperatorNamespace(),
+			fields.OneTermEqualSelector("metadata.name", name),
+		),
+		&v1.ConfigMap{},
+		0, // no resync period
+	)
+	_, err := informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(_, newObj interface{}) {
```

Comment:
What happens if the configmap is deleted ?

Reply:
Good question - I copied this code largely from existing CC code, which didn't have a deletion handler. I think in practice deleting this configmap will break the cluster, so it sort of doesn't matter what we do (and if it's deleted / re-created we'll get an ADD event which will be handled)